plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2260,16 +2262,25 @@ private void pruneAndDeleteStagingDirectories(List<DeclaredIntentionToWrite> dec
             }
         }
 
-        private void waitForAsyncRenames()
+        private void waitForAsyncFileSystemOperations()
         {
-            for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
-                getFutureValue(fileRenameFuture, TrinoException.class);
+            try {
+                fileSystemFutures.forEach(future -> getFutureValue(future, TrinoException.class));
+            }
+            catch (Throwable t) {
+                try {
+                    fileSystemFutures.forEach(future -> future.cancel(true));
```

Comment:
will `future.cancel` prevent unstarted future from starting (I think that what `cancel` was about)?

Reply:
yes, and it'll also stop the running futures with `CancellationException`.